packages/mdc-drawer/component.ts

```diff
@@ -0,0 +1,156 @@
+  static attachTo(root: Element): MDCDrawer {
+    return new MDCDrawer(root);
+  }
+
+  /**
+   * Returns true if drawer is in the open position.
+   */
+  get open(): boolean {
+    return this.foundation_.isOpen();
+  }
+
+  /**
+   * Toggles the drawer open and closed.
+   */
+  set open(isOpen: boolean) {
+    if (isOpen) {
+      this.foundation_.open();
+    } else {
+      this.foundation_.close();
+    }
+  }
+
+  private previousFocus_?: Element | null;
+  private scrim_!: HTMLElement | null; // assigned in initialSyncWithDOM()
+  private list_?: MDCList; // assigned in initialize()
+
+  private focusTrap_?: createFocusTrap.FocusTrap; // assigned in initialSyncWithDOM()
+  private focusTrapFactory_!: MDCDrawerFocusTrapFactory; // assigned in initialize()
+
+  private handleScrimClick_?: SpecificEventListener<'click'>; // initialized in initialSyncWithDOM()
+  private handleKeydown_!: SpecificEventListener<'keydown'>; // initialized in initialSyncWithDOM()
+  private handleTransitionEnd_!: SpecificEventListener<'transitionend'>; // initialized in initialSyncWithDOM()
+
+  initialize(
+    focusTrapFactory: MDCDrawerFocusTrapFactory = createFocusTrap as unknown as MDCDrawerFocusTrapFactory,
+    listFactory: MDCListFactory = (el) => new MDCList(el),
+  ) {
+    const listEl = this.root_.querySelector(`.${MDCListFoundation.cssClasses.ROOT}`);
+    if (listEl) {
+      this.list_ = listFactory(listEl);
+      this.list_.wrapFocus = true;
+    }
+    this.focusTrapFactory_ = focusTrapFactory;
+  }
+
+  initialSyncWithDOM() {
+    const {MODAL} = cssClasses;
+    const {SCRIM_SELECTOR} = strings;
+
+    this.scrim_ = (this.root_.parentNode as Element).querySelector<HTMLElement>(SCRIM_SELECTOR);
+
+    if (this.scrim_ && this.root_.classList.contains(MODAL)) {
+      this.handleScrimClick_ = () => (this.foundation_ as MDCModalDrawerFoundation).handleScrimClick();
+      this.scrim_.addEventListener('click', this.handleScrimClick_);
+      this.focusTrap_ = util.createFocusTrapInstance(this.root_ as HTMLElement, this.focusTrapFactory_);
+    }
+
+    this.handleKeydown_ = (evt) => this.foundation_.handleKeydown(evt);
+    this.handleTransitionEnd_ = (evt) => this.foundation_.handleTransitionEnd(evt);
+
+    this.listen('keydown', this.handleKeydown_);
+    this.listen('transitionend', this.handleTransitionEnd_);
+  }
+
+  destroy() {
+    this.unlisten('keydown', this.handleKeydown_);
+    this.unlisten('transitionend', this.handleTransitionEnd_);
+
+    if (this.list_) {
+      this.list_.destroy();
+    }
+
+    const {MODAL} = cssClasses;
+    if (this.scrim_ && this.handleScrimClick_ && this.root_.classList.contains(MODAL)) {
+      this.scrim_.removeEventListener('click', this.handleScrimClick_);
+      // Ensure drawer is closed to hide scrim and release focus
+      this.open = false;
+    }
+  }
+
+  getDefaultFoundation() {
+    // tslint:disable:object-literal-sort-keys
+    const adapter: MDCDrawerAdapter = {
```

Comment:
This looks right (has separate typed variable) but is missing the "do not inline" comment seen elsewhere.

Also in this boat: tab-indicator, tab-scroller, tabs/tab-bar, tabs/tab-bar-scroller, toolbar, top-app-bar

Reply:
Good catch! Not sure how I missed this. Fixed.